File: Unidad4/Programa10U5.java
package com.mycompany.programa10u5;
import java.util.Random;

public class Programa10U5 {

    public static void main(String[] args) {
        int longitud=8;
        String contrasena = aleatorio(longitud);
        System.out.println("Password: "+contrasena);
        
    }
    public static String aleatorio(int longitud){
        //declarar los caracteres permitidos
        String caracteres = "ABCDEFghijkl0123456789*-_.";
        Random random = new Random();//Sirve para generar numero
        //Costruir caracteres
        StringBuilder cadena = new StringBuilder(longitud);
        for(int i=0; i < longitud; i++){
            int indiceCaracter = random.nextInt(caracteres.length());
            cadena.append(caracteres.charAt(indiceCaracter));
        }
        return cadena.toString();
    }
}

File: Unidad 5 C#/Programa10U5.cs
using System;
using System.Text;

namespace programa10u5
{
    internal class Programa10U5
    {
        static void Main(string[] args)
        {
            int longitud = 8;
            string contrasena = GenerarContrasenaAleatoria(longitud);
            Console.WriteLine("Password: " + contrasena);
        }

        public static string GenerarContrasenaAleatoria(int longitud)
        {
            // Declarar los caracteres permitidos
            string caracteres = "ABCDEFghijkl0123456789*-_.";
            Random random = new Random();

            // Construir la cadena de caracteres aleatorios
            StringBuilder cadena = new StringBuilder(longitud);
            for (int i = 0; i < longitud; i++)
            {
                int indiceCaracter = random.Next(caracteres.Length);
                cadena.Append(caracteres[indiceCaracter]);
            }

            return cadena.ToString();
        }
    }
}
